docs/premium-features/custom-types-and-actions.md

```diff
@@ -0,0 +1,259 @@
+
+2. **Primitives**: Primitives are the basic data types supported by Amplication, such as String, Integer, Boolean, Date, and JSON. These primitives can be used as fields within your DTOs or as standalone types for your Actions.
+
+### Actions
+
+Actions in Amplication represent the API endpoints that define the operations available in your application. Each Action specifies the input, output, and other configurations specific to the API type (REST or GraphQL).
+
+For [REST API](/api/#rest-apis), an Action defines:
+
+- HTTP method (GET, POST, PUT, DELETE, etc.)
+- Path (e.g., `/users` or `/products/:id`)
+- Input parameters (query, path, or request body)
+- Output data structure (response body)
+
+For [GraphQL API](/api/#graphql-apis), an Action defines:
+
+- Operation type (Query or Mutation)
+- Input type (arguments)
+- Output type (return type)
+
+Actions can be of two types:
+
+1. **Default CRUD Actions**: Amplication automatically generates default CRUD (Create, Read, Update, Delete) Actions based on your application's entities. These Actions provide standard API endpoints for interacting with your data.
+
+   - For REST API, the default CRUD Actions typically include:
+     - `GET /entities` (Read all)
+     - `GET /entities/:id` (Read by ID)
+     - `POST /entities` (Create)
+     - `PUT /entities/:id` (Update)
+     - `DELETE /entities/:id` (Delete)
+
+   - For GraphQL API, the default CRUD Actions typically include:
+     - `entities` (Read all)
+     - `entity(id: ID!)` (Read by ID)
+     - `createEntity(input: CreateEntityInput!)` (Create)
+     - `updateEntity(id: ID!, input: UpdateEntityInput!)` (Update)
+     - `deleteEntity(id: ID!)` (Delete)
+
+2. **Custom Actions**: Custom Actions are user-defined API endpoints that extend the functionality of your application beyond the default CRUD operations. You can create Custom Actions to implement specific business logic, perform complex data transformations, or integrate with external services.
+
+## Accessing Custom Types & Actions
+
+To access the Custom Types & Actions feature, navigate to the new **APIs** tab in your Amplication dashboard.
+
+The APIs tab provides a centralized view of all your project's Modules, DTOs, and Actions. It's also further divided into sections for each Module, making it easy to browse and manage your project's custom types and actions.
+
+![APIs Tab Overview](./assets/custom-types-and-actions/api-tab-overview.png)
+
+## Modules
+
+Modules are organizational units that group related DTOs and Actions together. They help structure your application in a logical and maintainable way.
+
+Modules are closely tied to entities in your application. Each entity is generated as a Module containing all the DTO and API code related to that Module. You can also create additional Modules based on your business needs and requirements.
```

Comment:
I would rephrase "modules are closely tied to entities" to ensure people understand better it's not mandatory.
Perhaps something like - "A module is encapsulating a unit of business logic that is related to one application domain. By default, Amplication creates a module for each DB entity with all the CRUD operations, and you can add your own custom modules based on your business needs and requirements. You can add custom DTOs and custom Actions to a module - whether it's an entity-related module, or a custom module you've created"